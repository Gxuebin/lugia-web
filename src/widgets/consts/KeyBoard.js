/**
 * Created by liguoxin on 16/2/5.
 */

const Keys = {
  BACKSPACE: 8,

  TAB: 9,

  NUM_CENTER: 12,

  ENTER: 13,

  RETURN: 13,

  SHIFT: 16,

  CTRL: 17,
  CONTROL: 17,

  ALT: 18,

  PAUSE: 19,

  CAPS_LOCK: 20,

  ESC: 27,

  SPACE: 32,

  PAGEUP: 33,

  PAGEDOWN: 34,

  END: 35,

  HOME: 36,

  LEFT: 37,

  UP: 38,

  RIGHT: 39,

  DOWN: 40,

  PRINT_SCREEN: 44,

  INSERT: 45,

  DELETE: 46,

  ZERO: 48,

  ONE: 49,

  TWO: 50,

  THREE: 51,

  FOUR: 52,

  FIVE: 53,

  SIX: 54,

  SEVEN: 55,

  EIGHT: 56,

  NINE: 57,

  A: 65,

  B: 66,

  C: 67,

  D: 68,

  E: 69,

  F: 70,

  G: 71,

  H: 72,

  I: 73,

  J: 74,

  K: 75,

  L: 76,

  M: 77,

  N: 78,

  O: 79,

  P: 80,

  Q: 81,

  R: 82,

  S: 83,

  T: 84,

  U: 85,

  V: 86,

  W: 87,

  X: 88,

  Y: 89,

  Z: 90,

  CONTEXT_MENU: 93,

  NUM_ZERO: 96,

  NUM_ONE: 97,

  NUM_TWO: 98,

  NUM_THREE: 99,

  NUM_FOUR: 100,

  NUM_FIVE: 101,

  NUM_SIX: 102,

  NUM_SEVEN: 103,

  NUM_EIGHT: 104,

  NUM_NINE: 105,

  NUM_MULTIPLY: 106,

  NUM_PLUS: 107,

  NUM_MINUS: 109,

  NUM_PERIOD: 110,

  NUM_DIVISION: 111,

  F1: 112,

  F2: 113,

  F3: 114,

  F4: 115,

  F5: 116,

  F6: 117,

  F7: 118,

  F8: 119,

  F9: 120,

  F10: 121,

  F11: 122,

  F12: 123,
  FengHao: 186,
  JiaHao: 187,
  DouHao: 188,
  JianHao: 189,
  JuHao: 190,
  XieGang: 191,
  BLH: 192,
  ZKH: 219,
  SX: 220,
  YKH: 221,
  DYH: 222,
};

const safariKeys = {
  3: 13,
  63234: 37,
  63235: 39,
  63232: 38,
  63233: 40,
  63276: 33,
  63277: 34,
  63272: 46,
  63273: 36,
  63275: 35,
};

export function isNavKeyPress(e) {
  const k = normalizeKey(e.keyCode);
  return (k >= 33 && k <= 40) || k == Keys.RETURN || k == Keys.TAB || k == Keys.ESC;
}

export function isSpecialKey(e) {
  const k = normalizeKey(e.keyCode);
  return (
    (e.type == 'keypress' && e.ctrlKey) ||
    isNavKeyPress(e) ||
    k == Keys.BACKSPACE ||
    (k >= 16 && k <= 20) ||
    (k >= 44 && k <= 46)
  );
}

function normalizeKey() {
  return true;
}

export function keyCode2String(keyCode, shift, spaceStatus) {
  if (keyCode > 0 && keyCode < 128) {
    return String.fromCharCode(keyCode);
  }

  switch (keyCode) {
    case Keys.FengHao:
      return shift ? ':' : ';';
    default:
      return '';
  }
}

export default Keys;
