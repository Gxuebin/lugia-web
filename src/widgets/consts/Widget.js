/**
 *
 * create by ligx
 *
 * @flow
 */
export const Input = 'sv.widget.Input';
export const Align = 'sv.widget.Align';
export const Trigger = 'sv.widget.Trigger';
export const InputTag = 'sv.widget.InputTag';
export const Theme = 'sv.widget.Theme';
export const Menu = 'sv.widget.Menu';
export const DropMenu = 'sv.widget.DropMenu';
export const Tree = 'sv.widget.Tree';
export const ThrottleScroller = 'sv.widget.ThrottleScroller';
