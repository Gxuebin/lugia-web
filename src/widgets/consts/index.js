/**
 *
 * create by ligx
 *
 * @flow
 */
const Button = 'sv_widget_Button';
const Input = 'sv_widget_Input';
const Align = 'sv_widget_Align';
const Trigger = 'sv_widget_Trigger';
const Tooltip = 'sv_widget_Tooltip';
const ThemeWrapWidget = 'sv_widget_ThemeWrapWidget';

const InputTag = 'sv_widget_InputTag';
const InputTagCloseButton = 'sv_widget_InputTagCloseButton';
const InputTagFontItem = 'sv_widget_InputTagFontItem';
const InputTagFocuInput = 'sv_widget_InputTagFocuInput';
const InputTagMoreItem = 'sv_widget_InputTagMoreItem';
const ItemTagOption = 'sv_widget_ItemTagOption';
const InputTagItem = 'sv_widget_InputTagItem';
const InputTagClearButton = 'sv_widget_InputTagClearButton';

const Theme = 'sv_widget_Theme';
const Menu = 'sv_widget_Menu';
const MenuItem = 'sv_widget_MenuItem';
const Icon = 'sv_widget_Icon';
const AddIcon = 'sv_widget_AddIcon';
const RefershIcon = 'sv_widget_RefershIcon';
const CheckIcon = 'sv_widget_CheckIcon';
const ClearIcon = 'sv_widget_ClearIcon';
const DropMenu = 'sv_widget_DropMenu';
const Tree = 'sv_widget_Tree';
const TreeSelect = 'sv_widget_TreeSelect';
const TreeSelectLimitTitle = 'sv_widget_TreeSelectLimitTitle';
const ThrottleScroller = 'sv_widget_ThrottleScroller';

/* add by wangcuixia */
const Switch = 'sv_widget_Switch';
const Slider = 'sv_widget_Slider';
const SliderButton = 'sv_widget_Slider_Button';
const SliderIcon = 'sv_widget_Slider_Icon';

/* add by liangguodong */
const SearchIcon = 'sv_widget_SearchIcon';
const PullIcon = 'sv_widget_PullIcon';

const ButtonGroup = 'sv_widget_ButtonGroup';
const Radio = 'sv_widget_Radio';
const RadioGroup = 'sv_widget_RadioGroup';
const CheckBox = 'sv_widget_CheckBox';
const CheckBoxGroup = 'sv_widget_CheckBoxGroup';
const CheckButton = 'sv_widget_CheckButton';
const AmountInput = 'sv_widget_AmountInput';
const Col = 'sv_widget_Col';
const Row = 'sv_widget_Row';
const NumberInput = 'sv_widget_NumberInput';
const Divider = 'sv_widget_Divider';
const Panel = 'sv_widget_Panel';
const Collapse = 'sv_widget_Collapse';
const Avatar = 'sv_widget_Avatar';
const Badge = 'sv_widget_Badge';
const NumberTurn = 'sv_widget_NumberTurn';

/* add by szfeng */
const Select = 'sv_widget_Select';
const SelectLimitTitle = 'sv_widget_SelectLimitTitle';
const QueryInput = 'sv_widget_QueryInput';
const Tag = 'sv_widget_Tag';
const DropMenuButton = 'DropMenuButton';

const Layout = 'sv_widget_Layout';
const Header = 'sv_widget_Header';
const Content = 'sv_widget_Content';
const Footer = 'sv_widget_Footer';
const Aside = 'sv_widget_Aside';
const Affix = 'sv_widget_Affix';

const Tabs = 'sv_widget_Tabs';
const Tabpano = 'sv_widget_Tabpano';
const TabsHContainer = 'sv_widget_TabsHContainer';

export default {
  Button,
  Input,
  Align,
  Trigger,
  Tooltip,
  ThemeWrapWidget,
  InputTag,
  InputTagCloseButton,
  InputTagFontItem,
  InputTagFocuInput,
  InputTagMoreItem,
  ItemTagOption,
  InputTagItem,
  InputTagClearButton,
  Theme,
  Menu,
  MenuItem,
  Icon,
  AddIcon,
  RefershIcon,
  CheckIcon,
  ClearIcon,
  DropMenu,
  Tree,
  TreeSelect,
  TreeSelectLimitTitle,
  ThrottleScroller,
  Switch,
  Slider,
  SliderButton,
  SliderIcon,
  SearchIcon,
  PullIcon,
  Radio,
  RadioGroup,
  ButtonGroup,
  CheckBox,
  CheckBoxGroup,
  CheckButton,
  AmountInput,
  Col,
  Row,
  NumberInput,
  Divider,
  Panel,
  Collapse,
  Avatar,
  Badge,
  NumberTurn,
  Select,
  SelectLimitTitle,
  Layout,
  Header,
  Content,
  Footer,
  Aside,
  QueryInput,
  Tabs,
  Tabpano,
  TabsHContainer,
  Affix,
  Tag,
  DropMenuButton,
};
