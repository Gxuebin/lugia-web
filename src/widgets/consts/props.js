/**
 *
 * create by ligx
 *
 * @flow
 */
export const DisplayField = 'text';
export const ValueField = 'value';
