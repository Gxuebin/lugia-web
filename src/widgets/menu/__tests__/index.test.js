//@flow

import * as React from 'react';
import chai from 'chai';
import Menu from '../index';
import * as Widgets from '../../consts/Widget';
import * as Widget from '../../consts/Widget';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import Theme from '../../theme';

import Enzyme, { mount, render, } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter(), });

const { MenuItem, } = Menu;
const { expect: exp, } = chai;

describe('Menu', () => {


  it('DropMenu single selectedKeys 2', () => {
    expect(renderer.create(<Menu selectedKeys={['2', '3',]}>
      <MenuItem key="1">a</MenuItem>
      <MenuItem key="2">b</MenuItem>
      <MenuItem key="3">c</MenuItem>
      <MenuItem key="4">d</MenuItem>
    </Menu>).toJSON()).toMatchSnapshot();
  });
  it('DropMenu width: 500px height: 333px', () => {
    expect(renderer.create(<Theme config={{ [Widget.Menu]: { width: 550, height: 333, }, }}><Menu>
      <MenuItem key="1">a</MenuItem>
      <MenuItem key="2">b</MenuItem>
      <MenuItem key="3">c</MenuItem>
      <MenuItem key="4">d</MenuItem>
    </Menu></Theme>).toJSON()).toMatchSnapshot();
  });

  it('DropMenu mutliple selectedKeys 2', () => {
    expect(renderer.create((<Menu selectedKeys={['3', '4',]} mutliple>
      <MenuItem key="1">a</MenuItem>
      <MenuItem key="2">b</MenuItem>
      <MenuItem key="3">c</MenuItem>
      <MenuItem key="4">d</MenuItem>
    </Menu>)).toJSON()).toMatchSnapshot();
  });

  it('DropMenu mutliple onClick', () => {
    const checkedKey = '4';
    const dom = mount(<Menu mutliple>
      <MenuItem key="1">a</MenuItem>
      <MenuItem key="2">b</MenuItem>
      <MenuItem key="3">c</MenuItem>
      <MenuItem key={checkedKey} checked>d</MenuItem>
    </Menu>);
    dom.find(MenuItem).forEach(node => {
      exp(node.prop('checked')).to.be.false;
    });
    dom.find(MenuItem).forEach(node => node.simulate('click'));
    dom.find(MenuItem).forEach(node => {
      exp(node.prop('checked')).to.be.true;
    });
  });
  it('DropMenu mutliple onClick selectedKeys: 1 2', () => {
    const checkedKey = '4';
    const dom = mount(<Menu mutliple selectedKeys={['1', '2',]}>
      <MenuItem key="1">a</MenuItem>
      <MenuItem key="2">b</MenuItem>
      <MenuItem key="3">c</MenuItem>
      <MenuItem key={checkedKey} checked>d</MenuItem>
    </Menu>);
    exp(dom.find(MenuItem).at(0).prop('checked')).to.be.true;
    exp(dom.find(MenuItem).at(1).prop('checked')).to.be.true;
    exp(dom.find(MenuItem).at(2).prop('checked')).to.be.false;
    exp(dom.find(MenuItem).at(3).prop('checked')).to.be.false;
    dom.find(MenuItem).forEach(node => node.simulate('click'));
    exp(dom.find(MenuItem).at(0).prop('checked')).to.be.false;
    exp(dom.find(MenuItem).at(1).prop('checked')).to.be.false;
    exp(dom.find(MenuItem).at(2).prop('checked')).to.be.true;
    exp(dom.find(MenuItem).at(3).prop('checked')).to.be.true;
  });

  it('DropMenu single onClick selectedKeys: 1 2', () => {
    const checkedKey = '4';
    const dom = mount(<Menu selectedKeys={['1', '2',]}>
      <MenuItem key="1">a</MenuItem>
      <MenuItem key="2">b</MenuItem>
      <MenuItem key="3">c</MenuItem>
      <MenuItem key={checkedKey} checked>d</MenuItem>
    </Menu>);
    exp(dom.find(MenuItem).at(0).prop('checked')).to.be.false;
    exp(dom.find(MenuItem).at(1).prop('checked')).to.be.true;
    exp(dom.find(MenuItem).at(2).prop('checked')).to.be.false;
    exp(dom.find(MenuItem).at(3).prop('checked')).to.be.false;
    dom.find(MenuItem).forEach(node => node.simulate('click'));
    exp(dom.find(MenuItem).at(0).prop('checked')).to.be.false;
    exp(dom.find(MenuItem).at(1).prop('checked')).to.be.false;
    exp(dom.find(MenuItem).at(2).prop('checked')).to.be.false;
    exp(dom.find(MenuItem).at(3).prop('checked')).to.be.true;
  });

  it('compute see count', () => {
    const dom = mount(<Menu>
      <MenuItem key="1">1</MenuItem>
      <MenuItem key="2">2</MenuItem>
      <MenuItem key="3">3</MenuItem>
      <MenuItem key="4" checked>4</MenuItem>
      <MenuItem key="5" checked>5</MenuItem>
      <MenuItem key="6" checked>6</MenuItem>
      <MenuItem key="7" checked>7</MenuItem>
      <MenuItem key="8" checked>8</MenuItem>
      <MenuItem key="9" checked>9</MenuItem>
      <MenuItem key="11" checked>10</MenuItem>
      <MenuItem key="12" checked>11</MenuItem>
      <MenuItem key="13" checked>12</MenuItem>
      <MenuItem key="13" checked>13</MenuItem>
    </Menu>);

    exp(dom.find('li').length).to.be.equal(8);
  });

  it('children: items', () => {
    const items = [];
    for (let i = 0; i < 10; i++) {
      items.push(<MenuItem key={i}>{i}</MenuItem>);
    }
    const wraper = renderer.create(
      <Theme config={{ [Widget.Menu]: { width: 200, height: 250, }, }}>
        <Menu mutliple>
          {items}
        </Menu></Theme>);
    expect(wraper.toJSON()).toMatchSnapshot();
  });

  it('props: data ', () => {
    createMenuPropsDataCase({});
  });

  it('props: data & getPrefix & getSuffix ', () => {
    const items = [];
    for (let i = 0; i < 10; i++) {
      items.push({ key: `k${i}`, value: `v${i}`, });
    }
    const prefixItems = [];
    const getPrefix = item => {
      prefixItems.push(item);
      return <div className="lgx_prefix"></div>;
    };
    const suffixItems = [];
    const getSuffix = item => {
      suffixItems.push(item);
      return <div className="lgx_suffix"></div>;
    };
    const cmp = createMenuPropsDataCase({ getSuffix, getPrefix, });

    exp(cmp.find('.lgx_prefix').length).to.be.equal(10);
    exp(cmp.find('.lgx_suffix').length).to.be.equal(10);
    exp(prefixItems).to.be.eql(items);
    exp(suffixItems).to.be.eql(items);
  });
  it('props: data  & getSuffix ', () => {
    const items = [];
    for (let i = 0; i < 10; i++) {
      items.push({ key: `k${i}`, value: `v${i}`, });
    }
    const suffixItems = [];
    const getSuffix = item => {
      suffixItems.push(item);
      return <div className="lgx_suffix"></div>;
    };
    const cmp = createMenuPropsDataCase({ getSuffix, });

    exp(cmp.find('.lgx_suffix').length).to.be.equal(10);
  });

  it('props: data & getPrefix  ', () => {
    const items = [];
    for (let i = 0; i < 10; i++) {
      items.push({ key: `k${i}`, value: `v${i}`, });
    }
    const prefixItems = [];
    const getPrefix = item => {
      prefixItems.push(item);
      return <div className="lgx_prefix"></div>;
    };
    const cmp = createMenuPropsDataCase({ getPrefix, });

    exp(cmp.find('.lgx_prefix').length).to.be.equal(10);
    exp(prefixItems).to.be.eql(items);
  });


  function createMenuPropsDataCase (props: Object) {
    const items = [];
    for (let i = 0; i < 10; i++) {
      items.push({ key: `k${i}`, value: `v${i}`, });
    }
    const cmp = mount(<Theme config={{ [Widget.Menu]: { width: 200, height: 350, }, }}>
      <Menu data={items}  {...props}>
      </Menu></Theme>);
    exp(cmp.find(Widgets.MenuItem).length).to.be.equal(10);
    items.forEach(({ value, }, index) => {
      exp(cmp.find(Widgets.MenuItem).at(index).text()).to.be.equal(value);
    });
    return cmp;
  }
});
