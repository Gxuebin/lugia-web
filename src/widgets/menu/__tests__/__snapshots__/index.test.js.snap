// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu DropMenu mutliple selectedKeys 2 1`] = `
.c0 {
  outline: none;
  margin: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding-left: 0;
  list-style: none;
  height: 23.333333333333336em;
  max-height: 23.333333333333336em;
  overflow: hidden;
}

.c1 {
  box-sizing: border-box;
  position: relative;
  display: block;
  height: 2.916666666666667em;
  padding: 0.5833333333333334em 0.6666666666666667em;
  font-weight: 400;
  color: #333333;
  font-weight: 500;
  white-space: nowrap;
  cursor: pointer;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
}

.c1:hover {
  background-color: #edf0fe;
  font-weight: 900;
}

.c1::after {
  font-family: "sviconfont" !important;
  text-rendering: optimizeLegibility;
  content: "\\e73e";
  color: transparent;
  display: inline-block;
  font-size: 1.2rem;
  -webkit-transform: scale(.83333333) rotate(0deg);
  -ms-transform: scale(.83333333) rotate(0deg);
  transform: scale(.83333333) rotate(0deg);
  zoom: 1;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  right: 0.8333333333333334em;
  font-weight: 700;
  text-shadow: 0 0.1px 0,0.1px 0 0,0 -0.1px 0,-0.1px 0;
}

.c1:hover:after {
  color: #d0c8c8;
}

.c2 {
  box-sizing: border-box;
  position: relative;
  display: block;
  height: 2.916666666666667em;
  padding: 0.5833333333333334em 0.6666666666666667em;
  font-weight: 400;
  color: #684fff;
  font-weight: 900;
  white-space: nowrap;
  cursor: pointer;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
}

.c2:hover {
  background-color: #edf0fe;
  font-weight: 900;
}

.c2::after {
  font-family: "sviconfont" !important;
  text-rendering: optimizeLegibility;
  content: "\\e73e";
  color: transparent;
  display: inline-block;
  font-size: 1.2rem;
  -webkit-transform: scale(.83333333) rotate(0deg);
  -ms-transform: scale(.83333333) rotate(0deg);
  transform: scale(.83333333) rotate(0deg);
  zoom: 1;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  right: 0.8333333333333334em;
  font-weight: 700;
  text-shadow: 0 0.1px 0,0.1px 0 0,0 -0.1px 0,-0.1px 0;
}

.c2:after {
  color: #684fff;
}

.c2:hover:after {
  color: #684fff;
}

<ul
  className="c0"
>
  <li
    checked={false}
    className="c1"
    onClick={[Function]}
    title="a"
  >
    a
  </li>
  <li
    checked={false}
    className="c1"
    onClick={[Function]}
    title="b"
  >
    b
  </li>
  <li
    checked={true}
    className="c2"
    onClick={[Function]}
    title="c"
  >
    c
  </li>
  <li
    checked={true}
    className="c2"
    onClick={[Function]}
    title="d"
  >
    d
  </li>
</ul>
`;

exports[`Menu DropMenu single selectedKeys 2 1`] = `
.c0 {
  outline: none;
  margin: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding-left: 0;
  list-style: none;
  height: 23.333333333333336em;
  max-height: 23.333333333333336em;
  overflow: hidden;
}

.c1 {
  box-sizing: border-box;
  position: relative;
  display: block;
  height: 2.916666666666667em;
  padding: 0.5833333333333334em 0.6666666666666667em;
  font-weight: 400;
  color: #333333;
  font-weight: 500;
  white-space: nowrap;
  cursor: pointer;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
}

.c1:hover {
  background-color: #edf0fe;
  font-weight: 900;
}

.c2 {
  box-sizing: border-box;
  position: relative;
  display: block;
  height: 2.916666666666667em;
  padding: 0.5833333333333334em 0.6666666666666667em;
  font-weight: 400;
  color: #684fff;
  font-weight: 900;
  white-space: nowrap;
  cursor: pointer;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
}

.c2:hover {
  background-color: #edf0fe;
  font-weight: 900;
}

<ul
  className="c0"
>
  <li
    checked={false}
    className="c1"
    onClick={[Function]}
    title="a"
  >
    a
  </li>
  <li
    checked={false}
    className="c1"
    onClick={[Function]}
    title="b"
  >
    b
  </li>
  <li
    checked={true}
    className="c2"
    onClick={[Function]}
    title="c"
  >
    c
  </li>
  <li
    checked={false}
    className="c1"
    onClick={[Function]}
    title="d"
  >
    d
  </li>
</ul>
`;

exports[`Menu DropMenu width: 500px height: 333px 1`] = `
.c0 {
  width: 45.833333333333336em;
  outline: none;
  margin: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding-left: 0;
  list-style: none;
  height: 29.166666666666668em;
  max-height: 29.166666666666668em;
  overflow: hidden;
}

.c1 {
  box-sizing: border-box;
  position: relative;
  display: block;
  height: 2.916666666666667em;
  padding: 0.5833333333333334em 0.6666666666666667em;
  font-weight: 400;
  color: #333333;
  font-weight: 500;
  white-space: nowrap;
  cursor: pointer;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
}

.c1:hover {
  background-color: #edf0fe;
  font-weight: 900;
}

<span>
  <ul
    className="c0"
  >
    <li
      checked={false}
      className="c1"
      onClick={[Function]}
      title="a"
    >
      a
    </li>
    <li
      checked={false}
      className="c1"
      onClick={[Function]}
      title="b"
    >
      b
    </li>
    <li
      checked={false}
      className="c1"
      onClick={[Function]}
      title="c"
    >
      c
    </li>
    <li
      checked={false}
      className="c1"
      onClick={[Function]}
      title="d"
    >
      d
    </li>
  </ul>
</span>
`;

exports[`Menu children: items 1`] = `
.c3 {
  width: 16.666666666666668em;
  outline: none;
  margin: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding-left: 0;
  list-style: none;
  height: 23.333333333333336em;
  max-height: 23.333333333333336em;
  overflow: hidden;
}

.c4 {
  box-sizing: border-box;
  position: relative;
  display: block;
  height: 2.916666666666667em;
  padding: 0.5833333333333334em 0.6666666666666667em;
  font-weight: 400;
  color: #333333;
  font-weight: 500;
  white-space: nowrap;
  cursor: pointer;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
}

.c4:hover {
  background-color: #edf0fe;
  font-weight: 900;
}

.c4::after {
  font-family: "sviconfont" !important;
  text-rendering: optimizeLegibility;
  content: "\\e73e";
  color: transparent;
  display: inline-block;
  font-size: 1.2rem;
  -webkit-transform: scale(.83333333) rotate(0deg);
  -ms-transform: scale(.83333333) rotate(0deg);
  transform: scale(.83333333) rotate(0deg);
  zoom: 1;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  right: 0.8333333333333334em;
  font-weight: 700;
  text-shadow: 0 0.1px 0,0.1px 0 0,0 -0.1px 0,-0.1px 0;
}

.c4:hover:after {
  color: #d0c8c8;
}

.c1 {
  overflow: hidden;
  height: 20.833333333333336em;
  width: 16.666666666666668em;
  position: relative;
}

.c2 {
  height: 20.833333333333336em;
  width: 15.666666666666668em;
  position: absolute;
  display: inline-block;
}

.c0:hover .gaxbjm {
  opacity: 1;
}

.c5 {
  height: 20.833333333333336em;
  width: 15.666666666666668em;
  position: absolute;
  display: inline-block;
  left: 15.666666666666668em;
  width: 1em;
  opacity: 0;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
}

.c0:hover .c5 {
  opacity: 1;
}

.c6 {
  position: relative;
  background: #fbfbfb;
  width: 1.6666666666666667em;
  height: 25em;
  z-index: 996;
  width: 1em;
}

.c7 {
  position: absolute;
  top: 0;
  left: 0;
  cursor: pointer;
  text-align: center;
  border-radius: 0.6666666666666667em;
  background: #c2c2c2;
  color: #fff;
  font-size: 1.2rem;
  line-height: 2.5em;
  -webkit-transition: all 0.4s;
  transition: all 0.4s;
  width: 0.6666666666666667em;
  margin-left: 0.16666666666666669em;
  margin-right: 0.16666666666666669em;
}

.c7:hover {
  background-color: #7a7a7a;
}

<span>
  <div
    className="c0 c1"
    onWheel={[Function]}
  >
    <div
      className="c2"
    >
      <ul
        className="c3"
      >
        <li
          checked={false}
          className="c4"
          onClick={[Function]}
          title=""
        >
          0
        </li>
        <li
          checked={false}
          className="c4"
          onClick={[Function]}
          title=""
        >
          1
        </li>
        <li
          checked={false}
          className="c4"
          onClick={[Function]}
          title=""
        >
          2
        </li>
        <li
          checked={false}
          className="c4"
          onClick={[Function]}
          title=""
        >
          3
        </li>
        <li
          checked={false}
          className="c4"
          onClick={[Function]}
          title=""
        >
          4
        </li>
        <li
          checked={false}
          className="c4"
          onClick={[Function]}
          title=""
        >
          5
        </li>
        <li
          checked={false}
          className="c4"
          onClick={[Function]}
          title=""
        >
          6
        </li>
        <li
          checked={false}
          className="c4"
          onClick={[Function]}
          title=""
        >
          7
        </li>
      </ul>
    </div>
    <div
      className="c5"
    >
      <div
        className="c6"
        onMouseDown={[Function]}
        onMouseMove={[Function]}
        onMouseOut={[Function]}
        onMouseUp={[Function]}
        onWheel={[Function]}
        style={
          Object {
            "height": "20.833333333333336em",
          }
        }
      >
        <div
          className="c7"
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          style={
            Object {
              "height": "14.916666666666666em",
              "top": "0em",
            }
          }
        />
      </div>
    </div>
  </div>
</span>
`;
